pkg/controller/agent/hostnetwork/controller.go

```diff
@@ -0,0 +1,279 @@
+package hostnetwork
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	ctlharv1 "github.com/rancher/harvester/pkg/generated/controllers/harvester.cattle.io/v1alpha1"
+	"github.com/vishvananda/netlink"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/klog"
+
+	networkv1alpha1 "github.com/rancher/harvester-network-controller/pkg/apis/network.harvester.cattle.io/v1alpha1"
+	"github.com/rancher/harvester-network-controller/pkg/config"
+	"github.com/rancher/harvester-network-controller/pkg/controller/common"
+	"github.com/rancher/harvester-network-controller/pkg/generated/controllers/network.harvester.cattle.io/v1alpha1"
+	"github.com/rancher/harvester-network-controller/pkg/network"
+	"github.com/rancher/harvester-network-controller/pkg/network/iface"
+	"github.com/rancher/harvester-network-controller/pkg/network/vlan"
+)
+
+const (
+	controllerName = "harvester-network-hostnetwork-controller"
```

Comment:
add  project name to avoid finalizers conflict
```yaml
 finalizers:
  - wrangler.cattle.io/bridge-vlan-controller
  - wrangler.cattle.io/settings-controller
  - wrangler.cattle.io/harvester-network-settings-controller
  ```